lib/service_actor/result.rb

```diff
@@ -28,7 +28,7 @@ def success?
     end
 
     def failure?
-      super || false
+      self[:failure?] || false
```

Comment:
This is the part that prevented ruby 3.0 from working, I didn't find the root cause but with this change it works on all ruby versions

Reply:
Interesting that super should work differently, here 🤔. Fix is just as good, though!